components/constellation/constellation.rs

```diff
@@ -2180,58 +2177,32 @@ where
         }
     }
 
-    fn handle_request_wgpu_adapter(
-        &mut self,
-        source_pipeline_id: PipelineId,
-        browsing_context_id: BrowsingContextId,
-        request: FromScriptMsg,
-    ) {
-        let browsing_context_group_id = match self.browsing_contexts.get(&browsing_context_id) {
-            Some(bc) => &bc.bc_group_id,
-            None => return warn!("Browsing context not found"),
-        };
-        let host = match self
-            .pipelines
-            .get(&source_pipeline_id)
-            .map(|pipeline| &pipeline.url)
-        {
-            Some(ref url) => match reg_host(&url) {
-                Some(host) => host,
-                None => return warn!("Invalid host url"),
-            },
-            None => return warn!("ScriptMsg from closed pipeline {:?}.", source_pipeline_id),
+    fn handle_request_wgpu_adapter(&mut self, request: FromScriptMsg) {
+        let adapter_request = if let FromScriptMsg::RequestAdapter(sender, options, ids) = request {
+            WebGPURequest::RequestAdapter {
+                sender,
+                options,
+                ids,
+            }
+        } else {
+            return warn!("Wrong message type in handle_request_wgpu_adapter");
         };
```

Comment:
I don't understand the purpose of this change. If we only handle one message, we should make this method only accept the values we need from that message and make the caller deal with that. If we intend to accept more messages, we should accept an enum of only those valid messages.